### DISCUSSION TITLE
Coordinates Log while interactive True

### DISCUSSION THREAD

**Discussion by Simone100190** [marcomusy/vedo#387] [Q&A] [answered]
Hi! First and foremost thanks for this amazing project!
My problem is this:
I have a window where i render a 3D object from a STL mesh. I have (and must keep) interaction with the user so he/she can move around this mesh. While doing so, i want to record-log these movements in a file, so i can "replicate" the user interaction later on. I had a function "showCoords" which print out camera transformation matrix, position, focal point, view up, etc. I can call it with interaction False of course with a time.sleep(.1) and get a stream of coordinates, but obv the user cannot interact. I can also call it putting plotter.show(interactive=True) and showCoords in a while True statement, but i get the coordinates only if i press Enter (so, exiting de facto from the show statement).
How can i manage to get a constant stream of coordinates without pressing enter every time?

**Comment by marcomusy** [comment, marked as answer]
Hi Simone 
thanks for your interest in the library. While VTK offers a way to record the camera movements, it's not implemented in `vedo`, but I think you can still easily do it with a simple callback function following example `examples/basic/mouseclick.py`
Let me know if encounter any problems.

PS: if you 're keen on using native vtk commands you can leverage these 2 classes:
https://vtk.org/doc/nightly/html/classvtkCameraWidget.html#details
https://vtk.org/doc/nightly/html/classvtkCameraInterpolator.html

**Reply by Simone100190** [reply]
Oh my God you are fantastic! I've searched a bit in the examples but that one slipped under my radar completely.
Also, a couple of minutes response time? Fantastic! KUDOS KUDOS KUDOS!
